test/test_linalg.py

```diff
@@ -5488,7 +5499,7 @@ def run_subtest(matrix_size, batches, device, pivot, singular=False, a=None):
                 else:
                     rows, columns = matrix_size
                 if a is None:
-                    a = random_matrix(rows, columns, *batches, **dict(singular=singular, dtype=dtype)).to(device)
+                    a = random_matrix(rows, columns, *batches, **dict(singular=singular, dtype=dtype, device=device))
```

Comment:
Nice refactor

Reply:
It's actually a fix for `RuntimeError: svd: LAPACK library not found`. Because for CUDA and Meta devices it unconditionally tried to generate input using SVD on CPU and then transfer to the requested device.